packages/docs-reanimated/docs/guides/migration-from-3.x.md

```diff
@@ -8,18 +8,92 @@ sidebar_label: Migration from 3.x
+On the API surface level, Reanimated 4.x introduces only some minor renames and other breaking changes between 3.x and 4.x as listed below. All the animation logic you've written in Reanimated v2 or v3 API works in 4.x without little to no changes. Animations based on shared values will work the same way as before, simultaneously and interchangeably with CSS animations and transitions. This means you can adopt CSS animations and transitions in your codebase incrementally at your own pace.
 
-### Integration with other libraries
+### Added dependency on `react-native-worklets`
 
-All integrations with other third-party libraries such as [React Native Gesture Handler](https://docs.swmansion.com/react-native-gesture-handler/docs/) or [React Native Skia](https://shopify.github.io/react-native-skia/) work the same as on 3.x.
+In Reanimated 4, we finally decided to move [worklets](/docs/next/fundamentals/glossary#worklet) implementation to a separate npm package named `react-native-worklets`. You will need to install `react-native-worklets` package using your package manager and rebuild the native apps.
 
-### Migrating from 2.x
+### Renamed `react-native-reanimated/plugin`
+
+You will also need to update `babel.config.js` in your apps and change `'react-native-reanimated/plugin'` to `'react-native-worklets/plugin'`.
+
+### Changed the default behavior of `withSpring`
+
+`withSpring` animation no longer uses `restDisplacementThreshold` and `restSpeedThreshold` parameters, they were replaced with a single `energyThreshold` parameter.
+
+This parameter is relative to the animation, unlike absolute `restDisplacementThreshold` and `restSpeedThreshold`. This means that removing all usage of `restDisplacementThreshold` and `restSpeedThreshold` should be enough and you don't need to override `energyThreshold` in your codebase.
+
+`duration` parameter was changed that so it's now the perceptual duration of the animation, not the actual time it takes to complete. The actual time to complete is 1.5 times the perceptual duration. If you were using the `duration` parameter and want to achieve the same effect as before, you will need to divide the `duration` parameter by 1.5 in your codebase.
+
+The default parameters of the animation changed as well. They proved to be of little use in real-world applications. If you depended on them, you can still import them with the following import statement:
+
+```js
+import { Reanimated3DefaultSpringConfig } from 'react-native-worklets'; // For physics based defaults.
+import { Reanimated3DefaultSpringConfigWithDuration } from 'react-native-worklets'; // For duration based defaults.
+```
+
+You can explore the changes in the default parameters in [withSpring reference](/docs/next/animation/withSpring).
+
+### Removed `useWorkletCallback`
+
+`useWorkletCallback` was marked as deprecated in Reanimated 3 and removed in Reanimated 4.
+
+You will need to manually update all its usages in the following way:
```

Comment:
hm, for readers reading this guide it no longer be the future so I'd drop the future simple tense 
and the word `manually` seems daunting and here unnecessary 

```suggestion
To migrate from `useWorkletCallback`:
```